### PR TITLE
Cleanup and update to latest Keycloak (v21)

### LINKED ISSUE #6: replacement for keycloak >= 17
With an update to keycloak >= 17 we probably need to replace / rewrite this plugin:

```
In Keycloak 17 the default distribution is now powered by Quarkus, while the legacy WildFly powered distribution will still be around until June 2022 we highly recommend starting the migration as soon as possible.
```
https://www.keycloak.org/migration/migrating-to-quarkus

### PR DESCRIPTION
**Pull request by dnlkoch**
Please review @terrestris/devs.

Resolves #6.

### CODE REVIEW

File: src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java [hunk 1]
```diff
@@ -106,12 +108,10 @@ private void sendRequest(String stringEvent, boolean isAdminEvent) {
...
                 // Get response body
-                // TODO: Replace with logger
-                System.out.println(response.getStatusLine());
+                log.debug(response.getStatusLine());
             } catch(Exception e) {
-                // TODO: Replace with logger
-                System.out.println("UH OH!! " + e);
-                e.printStackTrace();
+                log.error("Error while requesting the SHOGun webhook " + e);
+                log.trace("Full stack trace: " + e.getMessage());
```
Comment (hwbllmnn) [inline src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java]:
```suggestion
                log.trace("Full stack trace: ", e);
```

File: src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java [hunk 2]
```diff
@@ -135,7 +135,7 @@ private String toString(Event event) {
             }
             return objectMapper.writeValueAsString(resultMap);
         } catch (JsonProcessingException e) {
-            System.out.printf("Could not serialize JSON: %s%n", e.getMessage());
+            log.error("Could not serialize JSON: " + e.getMessage());
```
Comment (hwbllmnn) [inline src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java:135]:
```suggestion
            log.error("Could not serialize JSON: ", e);
```

File: src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java [hunk 3]
```diff
@@ -156,13 +156,12 @@ private String toString(AdminEvent adminEvent) {
...
```
Comment (hwbllmnn) [inline src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java:157]:
```suggestion
            log.error("Could not serialize JSON: " + e.getMessage());
            log.trace("Stack trace:", e);
```

**Review by hwbllmnn — APPROVED** [review]

File: src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java (hunk 3 above)
Comment (dnlkoch) [reply src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java:157]: Fixed in 19756c5.

File: src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java (hunk 2 above)
Comment (dnlkoch) [reply src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java:135]: Fixed in 19756c5.

File: src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java (hunk 1 above)
Comment (dnlkoch) [reply src/main/java/de/terrestris/keycloak/providers/events/shogun/ShogunEventListenerProvider.java]: Fixed in 19756c5.

**Review by LukasLohoff — APPROVED** [review]
Nice! This will resolve https://github.com/terrestris/keycloak-event-listener-shogun/issues/6

Is keycoak >16 actually a requirement for this, or would the extension work with both jboss and quarkus-based keyloak images? if the former is the case, we should also include keycloak in the list of requirements

File: README.md [hunk 4]
```diff
@@ -1,54 +1,53 @@
...
 
-# Deploy
+# Installation
 
-* Copy target/event-listener-shogun-jar-with-dependencies.jar to {KEYCLOAK_HOME}/standalone/deployments
+Copy `target/event-listener-shogun-jar-with-dependencies.jar` to `{KEYCLOAK_HOME}/providers`.
 
-If you are working in a docker environment you might want to mount the `deployments` folder as a volume and copy the
-target to the host directory instead.
+If you are working in a [Docker environment](https://quay.io/repository/keycloak/keycloak) you might want to mount
```
Comment (simonseyock) [inline README.md:29]: There is also the official keycloak image: https://hub.docker.com/r/keycloak/keycloak, the quay.io one is only a mirror if I am not mistaken. They get configured the same way.

**Comment by dnlkoch** [comment]
> Is keycoak >16 actually a requirement for this, or would the extension work with both jboss and quarkus-based keyloak images? if the former is the case, we should also include keycloak in the list of requirements

Actually the extension should work with keycloak versions <21 since the interfaces `EventListenerProvider` and `EventListenerProviderFactory` didn't change between 16 and 21. But as defined in the [pom.xml](https://github.com/terrestris/keycloak-event-listener-shogun/pull/7/files#diff-9c5fb3d1b7e3b0f54bc5c4182965c4fe1f9023d449017cece3005d3f90e8e4d8R41) the dependencies are expected in the given Keycloak version.

**Comment by dnlkoch** [comment]
Thanks for the reviews!

**Comment by dnlkoch** [comment]
Thanks for the reviews!

**Pull request merged**